cookbook/integrations/flytekit_plugins/duckdb_examples/example.py

```diff
@@ -0,0 +1,157 @@
+    inputs=kwtypes(mydf=pd.DataFrame),
+)
+
+
+# %%
+# You can invoke the task from within a :py:func:`~flytekit:flytekit.workflow` and return both a Pandas dataframe and a PyArrow table.
+# The query will be executed on a Pandas dataframe, and the resulting output can belong to any StructuredDataset-compatible type.
+@task
+def get_pandas_df() -> pd.DataFrame:
+    return pd.DataFrame({"a": [1, 2, 3]})
+
+
+@workflow
+def pandas_wf() -> pd.DataFrame:
+    return simple_duckdb_query(mydf=get_pandas_df())
+
+
+@workflow
+def arrow_wf() -> pa.Table:
+    return simple_duckdb_query(mydf=get_pandas_df())
+
+
+if __name__ == "__main__":
+    print(f"Running pandas_wf()... {pandas_wf()}")
+    print(f"Running arrow_wf()... {arrow_wf()}")
+
+# %%
+# SQL query on Parquet file
+# =========================
+#
+# DuckDB enables direct querying of a parquet file without the need for intermediate conversions to a database.
+# If you wish to execute a SQL query on a parquet file stored in a public S3 bucket, you can use the ``httpfs`` library by installing and loading it.
+# Simply send the parquet file as a parameter to the ``SELECT`` query.
+parquet_duckdb_query = DuckDBQuery(
+    name="parquet_query",
+    query=[
+        "INSTALL httpfs",
+        "LOAD httpfs",
+        """SELECT hour(lpep_pickup_datetime) AS hour, count(*) AS count FROM READ_PARQUET(?) GROUP BY hour""",
+    ],
+    inputs=kwtypes(params=List[str]),
+)
+
+
+@workflow
+def parquet_wf(parquet_file: str) -> pd.DataFrame:
+    return parquet_duckdb_query(params=[parquet_file])
+
+
+if __name__ == "__main__":
+    parquet_file = (
+        "https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2022-02.parquet"
+    )
+    print(f"Running parquet_wf()... {parquet_wf(parquet_file=parquet_file)}")
+
+# %%
+# SQL query on StructuredDataset
+# ===============================
+#
+# To execute a SQL query on a structured dataset, you can simply run a query just like any other query on a Pandas dataframe or PyArrow table.
+sd_duckdb_query = DuckDBQuery(
+    name="sd_query",
+    query="SELECT * FROM sd_df WHERE i = 2",
+    inputs=kwtypes(sd_df=StructuredDataset),
+)
+
+
+@task
+def get_sd() -> StructuredDataset:
+    return StructuredDataset(
+        dataframe=pd.DataFrame.from_dict(
+            {"i": [1, 2, 3, 4], "j": ["one", "two", "three", "four"]}
+        )
+    )
+
+
+@workflow
+def sd_wf() -> pd.DataFrame:
+    sd_df = get_sd()
+    return sd_duckdb_query(sd_df=sd_df)
+
+
+if __name__ == "__main__":
+    print(f"Running sd_wf()... {sd_wf()}")
+
+# %%
+# Send parameters to multiple queries
+# ===================================
+#
+# To send parameters to multiple queries, use list of lists.
+#
+# .. note::
+#
+#   Sometimes, the annotation of parameter types can be somewhat complicated.
+#   In such situations, you can convert the list to a string using ``json.dumps``.
+#   The string will be automatically loaded into a list under the hood.
```

Comment:
can we say a little bit more about the behavior of sending a json list of parameters in this example?

For instance, the query list has 3 elements but the parameter list only has 2 elements. Can we explain how these two lists are aligned? I'm guessing they're aligned starting from the end of the list?